bigcode_eval/tasks/humaneval.py

```diff
@@ -57,17 +57,64 @@ def __init__(self, strip_prompt, k=[1, 10, 100], num_workers=16, timeout=3.0):
         self.num_workers = num_workers
         self.timeout = timeout
 
-    def get_dataset(self):
+        # Used for Nuggets experiments
+        self.one_shot = False
```

Comment:
I think it's confusing for get_dataset to also set some global data generation parameters here. Can we instead pass around a "dataset_spec" dataclass to any functions that actually need it?